mkosi

```diff
@@ -1027,8 +1027,14 @@ def reset_machine_id(args, workspace, run_build_script):
         os.unlink(machine_id)
         open(machine_id, "w+b").close()
         dbus_machine_id = os.path.join(workspace, 'root', 'var/lib/dbus/machine-id')
-        if os.path.exists(dbus_machine_id):
+        has_dbus_machine_id = False
+        try:
             os.unlink(dbus_machine_id)
+            has_dbus_machine_id = True
+        except FileNotFoundError:
+            pass
+
+        if has_dbus_machine_id:
```

Comment:
@fsateler no. the "else" branch of a "try" statement is only executed when no exception is thrown. The difference between the version you are now proposing and @keszybz's is that in your version  "os.symlink()" throwing FileNotFoundError is ignored, while in @keszybz's version it is not. And I think @keszybz' version is right here: we should only ignore the errors we really want to ignore, and that's the ones from os.unlink().